src/parcsr_ls/par_rap.c

```diff
@@ -34,81 +38,73 @@ hypre_BoomerAMGBuildCoarseOperatorKT( hypre_ParCSRMatrix  *RT,
    hypre_profile_times[HYPRE_TIMER_ID_RAP] -= hypre_MPI_Wtime();
 #endif
 
-   MPI_Comm        comm = hypre_ParCSRMatrixComm(A);
+   MPI_Comm               comm = hypre_ParCSRMatrixComm(A);
 
-   HYPRE_MemoryLocation memory_location_RAP = hypre_ParCSRMatrixMemoryLocation(A);
+   HYPRE_MemoryLocation   memory_location_RAP = hypre_ParCSRMatrixMemoryLocation(A);
 
-   hypre_CSRMatrix *RT_diag = hypre_ParCSRMatrixDiag(RT);
-   hypre_CSRMatrix *RT_offd = hypre_ParCSRMatrixOffd(RT);
-   HYPRE_Int             num_cols_diag_RT = hypre_CSRMatrixNumCols(RT_diag);
-   HYPRE_Int             num_cols_offd_RT = hypre_CSRMatrixNumCols(RT_offd);
-   HYPRE_Int             num_rows_offd_RT = hypre_CSRMatrixNumRows(RT_offd);
+   hypre_CSRMatrix       *RT_diag = hypre_ParCSRMatrixDiag(RT);
+   hypre_CSRMatrix       *RT_offd = hypre_ParCSRMatrixOffd(RT);
+   HYPRE_Int              num_cols_diag_RT = hypre_CSRMatrixNumCols(RT_diag);
+   HYPRE_Int              num_cols_offd_RT = hypre_CSRMatrixNumCols(RT_offd);
+   HYPRE_Int              num_rows_offd_RT = hypre_CSRMatrixNumRows(RT_offd);
    hypre_ParCSRCommPkg   *comm_pkg_RT = hypre_ParCSRMatrixCommPkg(RT);
-   HYPRE_Int             num_recvs_RT = 0;
-   HYPRE_Int             num_sends_RT = 0;
-   HYPRE_Int             *send_map_starts_RT;
+   HYPRE_Int              num_recvs_RT = 0;
+   HYPRE_Int              num_sends_RT = 0;
+   HYPRE_Int             *send_map_starts_RT = NULL;
    HYPRE_Int             *send_map_elmts_RT;
 
-   hypre_CSRMatrix *A_diag = hypre_ParCSRMatrixDiag(A);
-
-   HYPRE_Real      *A_diag_data = hypre_CSRMatrixData(A_diag);
+   hypre_CSRMatrix       *A_diag = hypre_ParCSRMatrixDiag(A);
+   HYPRE_Complex         *A_diag_data = hypre_CSRMatrixData(A_diag);
```

Comment:
Do we really want to convert all these to Complex? Does this make sense?

Reply:
The new code makes sense to me because hypre_CSRMatrixData(A) is a `HYPRE_Complex`:

https://github.com/hypre-space/hypre/blob/61fa167d71286825c16c0634f832d8d96edd4030/src/seq_mv/csr_matrix.h#L64

https://github.com/hypre-space/hypre/blob/61fa167d71286825c16c0634f832d8d96edd4030/src/seq_mv/csr_matrix.h#L45

Reply:
I realize it works because HYPRE_Complex is generally defined as HYPRE_Real, but it is kind of a mess, since we really don't work with complex numbers, and they don't always make sense. This was mostly converted to Complex for the QCD guys so they could use CG or so. That is why some numbers are complex and others are not.

Reply:
OK, I see the point now. In this function at least, the use of complex numbers seems to make sense, i.e., `HYPRE_Complex` could be different than `HYPRE_Real` and this function would still work.